wondrous-app/components/Common/Sidebar/Entity/List.tsx

```diff
@@ -0,0 +1,168 @@
+    (link, type = null) =>
+    () => {
+      if (type && setEntityType) {
+        setEntityType(type);
+        return;
+      }
+      router.push(link);
+    };
+  const link = orgBoard ? `/organization/${board?.orgData?.username}` : `/pod/${board?.podId}`;
+  const sidebarCount = useSidebarCount();
+  return {
+    handleOnClick,
+    data: [
+      {
+        label: 'Workspaces',
+        items: [
+          {
+            text: 'Tasks',
+            Icon: CheckBoxIcon,
+            link: `${link}/boards?entity=${ENTITIES_TYPES.TASK}`,
+            count: sidebarCount.taskCount,
+            entityType: ENTITIES_TYPES.TASK,
+          },
+          {
+            text: 'Bounties',
+            Icon: StartIcon,
+            link: `${link}/boards?entity=${ENTITIES_TYPES.BOUNTY}`,
+            count: sidebarCount.bountyCount,
+            entityType: ENTITIES_TYPES.BOUNTY,
+          },
+          {
+            text: 'Milestones',
+            Icon: FlagIcon,
+            link: `${link}/boards?entity=${ENTITIES_TYPES.MILESTONE}`,
+            count: sidebarCount.milestoneCount,
+            entityType: ENTITIES_TYPES.MILESTONE,
+          },
+          {
+            text: 'Proposals',
+            Icon: ContentPaste,
+            link: `${link}/boards?entity=${ENTITIES_TYPES.PROPOSAL}`,
+            count: sidebarCount.proposalCount,
+            entityType: ENTITIES_TYPES.PROPOSAL,
+          },
+          // {
+          //   text: 'Pods',
+          //   Icon: PodIcon,
+          //   link: null, // link: not sure yet
+          // },
+        ],
+      },
+      {
+        label: 'Community',
+        items: [
+          {
+            text: 'Analytics',
+            Icon: PieChartIcon,
+            link: `${link}/analytics`,
+          },
+          {
+            text: 'Activity',
+            Icon: ShowChartIcon,
+            link: `${link}/activities`,
+          },
+          {
+            text: 'Members',
+            Icon: GroupIcon,
+            link: `${link}/members`,
+            count: sidebarCount.membersCount,
+          },
+          {
+            text: 'Roles',
+            Icon: StackIcon,
+            link: {
+              pathname: router.pathname,
+              query: {
+                ...router.query,
+                roles: true,
+              },
+            },
+            count: sidebarCount.rolesCount,
+          },
+
+          {
+            text: 'Resources',
+            Icon: FolderIcon,
+            link: `${link}/docs`,
+            count: sidebarCount.resourcesCount,
+          },
+        ],
+      },
+    ],
+  };
+};
+
+const location = () => {
+  if (typeof window !== 'undefined') return window.location.pathname + window.location.search;
+  return '';
+};
+
+const List = () => {
+  const { data, handleOnClick } = useSidebarData();
+  const router = useRouter();
+  const isActive = (entityType, link) => (entityType ? location().includes(link) : router.asPath.includes(link));
```

Comment:
if we already have the router instead of location() can we use router.asPath?

Reply:
@Lamperoyge No, we can't. The `router` context is not updated when we use `setEntityType`.